bin/update_cucumber

```diff
@@ -77,13 +80,15 @@ class {
             && ($repository['package']['name'] ?? null) === 'cucumber/gherkin-monorepo'
             && isset($repository['package']['version'])
             && is_array($repository['package']['source'] ?? null)
-            && isset($repository['package']['source']['reference']),
+            && isset($repository['package']['source']['reference'])
+            && is_string($repository['package']['source']['reference']),
             'First composer.json repository should be our custom gherkin package with expected structure',
         );
 
         $composerVersion = $repository['package']['version'];
         assert(
-            is_array($composerConfig['require-dev'] ?? null)
+            is_string($composerVersion)
```

Comment:
Does this matter? If it's not a string it won't `===` the value at the end of this assert anyway?

Reply:
Yes, PHPStan is surprisingly thorough!

Here's why:
- it was not asserting that `$composerVersion` (=> therefore it's defined as mixed)
- it didn't assert that `$composerConfig['require-dev']['cucumber/gherkin-monorepo']` is a string (=> therefore also seen as mixed)
- it's possible that `mixed === mixed` (e.g. arrays)
- later one we concatenate `$composerVersion` which at that point is still considered `mixed`